modules/rosa-node-drain-grace-period.adoc

```diff
@@ -0,0 +1,83 @@
+// Module included in the following assemblies:
+//
+// * rosa_cluster_admin/rosa_nodes/rosa-managing-worker-nodes.adoc
+//this module applies to ROSA HCP only
+
+:_mod-docs-content-type: PROCEDURE
+[id="rosa-node-drain-grace-period_{context}"]
+= Configuring node drain grace periods 
+
+You can configure the node drain grace period for machine pools in your cluster. The node drain grace period for a machine pool is how long the cluster respects the Pod Disruption Budget protected workloads when upgrading or replacing the machine pool. After this grace period, all remaining workloads will be forcibly evicted. The value range for the node drain grace period is from 0 to 1 week. With the default value 0, or empty value, the machine pool drains without any time limitation until complete.
+
+
+.Prerequisites
+
+* You installed and configured the latest {product-title} (ROSA) CLI, `rosa`, on your workstation.
+* You logged in to your Red Hat account by using the ROSA CLI.
+* You created a {hcp-title-first} cluster.
+* You have an existing machine pool.
+
+.Procedure
+
+. List all of the machine pools in the cluster:
++
+[source,terminal]
+----
+$ rosa list machinepools --cluster=<cluster_name>
+----
++
+.Example output
+[source,terminal]
+----
+ID          AUTOSCALING  REPLICAS  INSTANCE TYPE [...] AVAILABILITY ZONES  SUBNET  VERSION  AUTOREPAIR  TUNING CONFIGS
+workers      No           2         m5.xlarge    [...] us-east-1a          N/A     4.14.18  Yes
+db-nodes-mp  No           2         m5.xlarge    [...] us-east-1a          No      4.14.18  Yes
+----
+
+. Check the node drain grace period for a machine pool:
++
+[source,terminal]
+----
+$ rosa describe machinepool --cluster <cluster_name> <machinepool_name>
+----
++
+.Example output
+[source,terminal]
+----
+ID:                                    workers
+Cluster ID:                            2a90jdl0i4p9r9k9956v5ocv40se1kqs       
+Node drain grace period:               <1>
+----
++
+<1> If this value is empty, the machine pool drains without any time limitation until complete.
+
+. Optional: Update the node drain grace period for a machine pool:
++
+[source,terminal]
+----
+$ rosa edit machinepool --node-drain-grace-period="<node_drain_grace_period_value>" --cluster=<cluster_name>  <machinepool_name>
+----
++
+[NOTE]
+====
+You cannot change the node drain grace period after a machine pool upgrade has started.
```

Comment:
Not sure how to word this to mean that it's possible to change the node drain grace period while an upgrade is in progress, but it won't be used in the current upgrade, but will be effective for future upgrades.

Reply:
Fixed